services/speech_to_text/v1.js

```diff
@@ -74,121 +215,102 @@ module.exports = function (RED) {
 
         node.error(message_err_band, msg)
         return;
-      }
+      }  
 
+      // Has to be there, as its a checkbox, but flows switching from the old (frankly 
+      // unbeliveable) select might not have it set.
       if (!config.continuous) {
         var message_err_continuous = 'Missing continuous details, unable to process speech.';
 
         node.error(message_err_continuous, msg)
         return;
-      }
+      }  
 
-      var model = config.lang + '_' + config.band;
+      // The input comes in on msg.payload, and can either be an audio file or a string 
+      // representing a URL.  
+      if (!msg.payload instanceof Buffer || !typeof msg.payload === 'string') {
+        var message = 'Invalid property: msg.payload, can only be a URL or a Buffer.';
 
-      var speech_to_text = watson.speech_to_text({
-        username: username,
-        password: password,
-        version: 'v1',
-        url: 'https://stream.watsonplatform.net/speech-to-text/api'
-      });
+        node.error(message, msg)
+        return;
+      }
 
-      var s2t = function (audio, format, cb) {
-        if (format !== 'wav' && format !== 'flac' && format !== 'ogg') {
-          var message_err_format = 'Audio format (' + format + ') not supported, must be encoded as WAV, FLAC or OGG.';
+      // This check is repeated just before the call to the service, but 
+      // its also performed here as a double check. 
+      if (!(msg.payload instanceof Buffer)) {
+        if (typeof msg.payload === 'string' && !urlCheck(msg.payload)) {
+          var message = 'Invalid URL.';
 
-          node.error(message_err_format, msg)
+          node.error(message, msg)
           return;
         }
+      } else {
+        var f = fileType(msg.payload).ext;
+        switch (f) {
+        case 'wav':
+        case 'flac':
+        case 'ogg':
+          break;
+        default:
+          var message_err_format = 'Audio format (' + f + ') not supported, must be encoded as WAV, FLAC or OGG.';
 
-        if (format === 'ogg') { 
-          format += ';codecs=opus';
-        }
-
-        var params = {};
-        
-        if(config.continuous === 'true')
-        {
-          params = {
-            audio: audio,
-            content_type: 'audio/' + format,
-            model: model,
-            continuous: true
-          };
-
-        } else {
-          params = {
-            audio: audio,
-            content_type: 'audio/' + format,
-            model: model,
-            continuous: false
-          };
-
+          node.error(message_err_format, msg)
+          return;  
         }
+      }  
 
-
-        node.status({fill:'blue', shape:'dot', text:'requesting'});
-        speech_to_text.recognize(params, function (err, res) {
-          node.status({})
+      // We are now ready to process the input data 
+      // If its a buffer then need to read it all before invoking the service 
+      var params = {};
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["params" is defined but never used](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2636061996/issues/source?bid=3225115&fileBranchId=3319654#l268)